lib/modules/platform/gerrit/utils.ts

```diff
@@ -0,0 +1,148 @@
+  }
+  if (!(opts.username && opts.password)) {
+    throw new Error(
+      'Init: You must configure a Gerrit Server username/password'
+    );
+  }
+  url.username = opts.username;
+  url.password = opts.password;
+  url.pathname = joinUrlParts(
+    url.pathname,
+    'a',
+    encodeURIComponent(repository)
+  );
+  logger.trace(
+    { url: url.toString() },
+    'using URL based on configured endpoint'
+  );
+  return url.toString();
+}
+
+export function mapPrStateToGerritFilter(state?: PrState): string {
+  switch (state) {
+    case 'closed':
+      return 'status:closed';
+    case 'merged':
+      return 'status:merged';
+    case '!open':
+      return '-status:open';
+    case 'open':
+      return 'status:open';
+    case 'all':
+    default:
+      return '-is:wip';
+  }
+}
+
+export function mapGerritChangeToPr(change: GerritChange): Pr {
+  return {
+    number: change._number,
+    state: mapGerritChangeStateToPrState(change.status),
+    sourceBranch: extractSourceBranch(change) ?? change.branch,
+    targetBranch: change.branch,
+    title: change.subject,
+    reviewers:
+      change.reviewers?.REVIEWER?.map((reviewer) => reviewer.username!) ?? [],
+    bodyStruct: {
+      hash: hashBody(findPullRequestBody(change)),
+    },
+  };
+}
+
+export function mapGerritChangeStateToPrState(
+  state: GerritChangeStatus
+): PrState {
+  switch (state) {
+    case 'NEW':
+      return 'open';
+    case 'MERGED':
+      return 'merged';
+    case 'ABANDONED':
+      return 'closed';
+  }
+  return 'all';
+}
+export function extractSourceBranch(change: GerritChange): string | undefined {
+  return change.hashtags
+    ?.find((tag) => tag.startsWith('sourceBranch-'))
+    ?.replace('sourceBranch-', '');
+}
+
+export function findPullRequestBody(change: GerritChange): string | undefined {
+  const msg = change.messages?.findLast(
+    (msg) => msg.tag === TAG_PULL_REQUEST_BODY
+  );
+  if (msg) {
+    return msg.message.replace(/^Patch Set \d+:\n\n/, ''); //TODO: check how to get rid of the auto-added prefix?
+  }
+  return undefined;
+}
+
+export function mapBranchStatusToLabel(
+  state: BranchStatus,
+  label: GerritLabelTypeInfo
+): number {
+  const numbers = Object.keys(label.values).map((x) => parseInt(x, 10));
+  switch (state) {
+    case 'green':
+      return Math.max(...numbers);
+    case 'yellow':
+    case 'red':
+      return Math.min(...numbers);
+  }
+  // istanbul ignore next
+  return label.default_value;
+}
+
+export function mapBranchStateContextToLabel(
+  context: string | null | undefined,
+  labelMapping: RepoGlobalConfig['gerritLabelMapping'],
+  gerritLabels: Record<string, GerritLabelTypeInfo>
+): {
+  labelName?: string;
+  label?: GerritLabelTypeInfo;
+} {
+  let labelName;
+  switch (context) {
+    //TODO #21442
+    case 'renovate/stability-days':
+      //not usefully for Gerrit, Renovate only creates the change after the minimumReleaseAge has been exceeded. That way the label would always be green...
```

Comment:
Wrong. It depends on `internalChecksFilter`[^1]

[^1]: https://docs.renovatebot.com/configuration-options/#internalchecksfilter